festim/boundary_conditions/flux_bc.py

```
@@ -0,0 +1,149 @@
+        expression of the function is stored in self.bc_expr.
+
+        Args:
+            mesh (dolfinx.mesh.Mesh) : the mesh
+            function_space (dolfinx.fem.FunctionSpace): the function space
+            temperature (float): the temperature
+            t (dolfinx.fem.Constant): the time
+        """
+        x = ufl.SpatialCoordinate(mesh)
+
+        if isinstance(self.value, (int, float)):
+            self.value_fenics = F.as_fenics_constant(mesh=mesh, value=self.value)
+
+        elif callable(self.value):
+            arguments = self.value.__code__.co_varnames
+
+            if "t" in arguments and "x" not in arguments and "T" not in arguments:
+                # only t is an argument
+                if not isinstance(self.value(t=float(t)), (float, int)):
+                    raise ValueError(
+                        f"self.value should return a float or an int, not {type(self.value(t=float(t)))} "
+                    )
+                self.value_fenics = F.as_fenics_constant(
+                    mesh=mesh, value=self.value(t=float(t))
+                )
+            else:
+                self.value_fenics = fem.Function(function_space)
+                kwargs = {}
+                if "t" in arguments:
+                    kwargs["t"] = t
+                if "x" in arguments:
+                    kwargs["x"] = x
+                if "T" in arguments:
+                    kwargs["T"] = temperature
+
+                # store the expression of the boundary condition
+                # to update the value_fenics later
+                self.bc_expr = fem.Expression(
+                    self.value(**kwargs),
+                    function_space.element.interpolation_points(),
+                )
+                self.value_fenics.interpolate(self.bc_expr)
+
+    def update(self, t):
+        """Updates the source value
+
+        Args:
+            t (float): the time
+        """
+        if callable(self.value):
```

Is this needed? I would replace by `if self.time_dependent`? what if it isn't? should we raise an error? or just a print?

---

This is how we also handle it in sources, and Dirichlet bc, so maybe we could change this for all of them in another PR?

---

Sounds good with me

---

I opened an issue #743 to remind us of this